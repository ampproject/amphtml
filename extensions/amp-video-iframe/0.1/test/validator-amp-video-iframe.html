<!--
  Test Description:
  Tests for amp-video-iframe tag.
-->
<!doctype html>
<html âš¡ lang="en">
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-video-iframe" src="https://cdn.ampproject.org/v0/amp-video-iframe-latest.js"></script>
</head>
<body>
  <!-- Valid -->
  <amp-video-iframe
      width="640"
      height="360"
      layout="responsive"
      src="https://foo.com">
  </amp-video-iframe>

  <!-- Valid -->
  <amp-video-iframe
      width="640"
      height="360"
      layout="responsive"
      src="https://foo.com"
      poster="images/kitten-playing.png">
  </amp-video-iframe>

  <!-- Invalid: Incorrect attribute value for autoplay -->
  <amp-video-iframe
      autoplay=true
      layout=fill
      width=300
      height=500
      src="https://foo.com"
      poster="images/kitten-playing.png">
  </amp-video-iframe>

  <!-- Valid: correct attribute value for autoplay -->
  <amp-video-iframe
      autoplay
      layout=fill
      width=300
      height=500
      src="https://foo.com"
      poster="images/kitten-playing.png">
  </amp-video-iframe>

  <!-- Valid: Correct attribute value for rotate-to-fullscreen -->
  <amp-video-iframe
      rotate-to-fullscreen
      layout=fill
      width=300
      height=500
      src="https://foo.com"
      poster="images/kitten-playing.png">
  </amp-video-iframe>

  <!-- Invalid: Incorrect attribute value for rotate-to-fullscreen -->
  <amp-video-iframe
      rotate-to-fullscreen=true
      layout=fill
      width=300
      height=500
      src="https://foo.com"
      poster="images/kitten-playing.png">
  </amp-video-iframe>
</body>
</html>
