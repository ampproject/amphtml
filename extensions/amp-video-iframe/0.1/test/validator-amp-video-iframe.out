FAIL
|  <!--
|    Copyright 2018 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests for amp-video-iframe tag.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <link rel="canonical" href="./regular-html-version.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|    <script async custom-element="amp-video-iframe" src="https://cdn.ampproject.org/v0/amp-video-iframe-latest.js"></script>
|  </head>
|  <body>
|    <!-- Valid -->
|    <amp-video-iframe
|        width="640"
|        height="360"
|        layout="responsive"
|        src="https://foo.com">
|    </amp-video-iframe>
|
|    <!-- Valid -->
|    <amp-video-iframe
|        width="640"
|        height="360"
|        layout="responsive"
|        src="https://foo.com"
|        poster="images/kitten-playing.png">
|    </amp-video-iframe>
|
|    <!-- Invalid: Incorrect attribute value for autoplay -->
|    <amp-video-iframe
>>   ^~~~~~~~~
amp-video-iframe/0.1/test/validator-amp-video-iframe.html:49:2 The attribute 'autoplay' in tag 'amp-video-iframe' is set to the invalid value 'true'. (see https://amp.dev/documentation/components/amp-video-iframe/)
|        autoplay=true
|        layout=fill
|        width=300
|        height=500
|        src="https://foo.com"
|        poster="images/kitten-playing.png">
|    </amp-video-iframe>
|
|    <!-- Valid: correct attribute value for autoplay -->
|    <amp-video-iframe
|        autoplay
|        layout=fill
|        width=300
|        height=500
|        src="https://foo.com"
|        poster="images/kitten-playing.png">
|    </amp-video-iframe>
|
|    <!-- Valid: Correct attribute value for rotate-to-fullscreen -->
|    <amp-video-iframe
|        rotate-to-fullscreen
|        layout=fill
|        width=300
|        height=500
|        src="https://foo.com"
|        poster="images/kitten-playing.png">
|    </amp-video-iframe>
|
|    <!-- Invalid: Incorrect attribute value for rotate-to-fullscreen -->
|    <amp-video-iframe
>>   ^~~~~~~~~
amp-video-iframe/0.1/test/validator-amp-video-iframe.html:79:2 The attribute 'rotate-to-fullscreen' in tag 'amp-video-iframe' is set to the invalid value 'true'. (see https://amp.dev/documentation/components/amp-video-iframe/)
|        rotate-to-fullscreen=true
|        layout=fill
|        width=300
|        height=500
|        src="https://foo.com"
|        poster="images/kitten-playing.png">
|    </amp-video-iframe>
|  </body>
|  </html>
